internal/controller/controlplane/util.go

```diff
@@ -88,3 +89,20 @@ func (c *K0sController) createKubeconfigSecret(ctx context.Context, cfg *api.Con
 
 	return c.Create(ctx, kcSecret)
 }
+
+func (c *K0sController) getKubeClient(ctx context.Context, cluster *clusterv1.Cluster) (*kubernetes.Clientset, error) {
+	data, err := kubeconfig.FromSecret(ctx, c.Client, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.Name})
+	if err != nil {
+		return nil, err // TODO
+	}
+	config, err := clientcmd.NewClientConfigFromBytes(data)
+	if err != nil {
+		return nil, err // TODO
```

Comment:
```suggestion
		return nil, fmt.Errorf("Error generating %s clientconfig: %w", cluster.Name, err)
```

So that we can debug a bit better.

Reply:
Done